infra/src/dsl/dsl.ts

```diff
@@ -171,7 +172,10 @@ export class ServiceBuilder<ServiceType> {
    */
   initContainer(ic: Optional<InitContainers, 'envs' | 'secrets' | 'features'>) {
     if (ic.postgres) {
-      ic.postgres = this.withDefaults(ic.postgres)
+      ic.postgres = {
+        ...this.withDefaults(ic.postgres),
+        extensions: ic.postgres.extensions,
```

Comment:
I feel like we shouldn't need to specify extenstions specifically when making the object, as it is set in withDefaults?

Reply:
Removing this breaks the tests though, so I'm confused.